### DISCUSSION TITLE
Strange updateLineContent Behaviour

### DISCUSSION THREAD

**Discussion by para-dise** [Minestom/Minestom#820] [Help] [answered]
Hi,

So I'm trying to update the contents of a sidebar

I first created the sidebar line with

```java
        this.sidebar.createLine(
                new Sidebar.ScoreboardLine(
                        "some_line_2",
                        LegacyComponentSerializer.legacyAmpersand().deserialize("&dWAITING FOR GAME TO START..."),
                        2
                )
        );
```

And later on, I try to update it with
```java
this.sidebar.updateLineContent("some_line_2", LegacyComponentSerializer.legacyAmpersand().deserialize("&d" + turnOrder.get(0).getName()));
```

However, it looks like the following:
![](https://i.imgur.com/YbxEQ0o.png)

Am I doing something wrong here?

**Comment by Moulberry** [comment, marked as answer]
Is `turnOrder.get(0).getName()` a component? If so, `"&d" + turnOrder.get(0).getName()` will concatenate &d with Component#toString(), giving the output you see.

What you probably want to do instead is: `turnOrder.get(0).getName().color(NamedTextColor.LIGHT_PURPLE)`, no jank LegacyComponentSerializer necessary
